cool-core/src/main/java/com/nus/cool/core/cohort/refactor/CohortProcessor.java

```diff
@@ -90,10 +116,62 @@ public CohortRet process(CubeRS cube) throws IOException {
     this.tuple = new ProjectedTuple(this.projectedSchemaSet);
     for (CubletRS cublet : cube.getCublets()) {
       processCublet(cublet);
+
+      // record result user id list
+      for (Map.Entry<String, List<String>> ele : this.result.getCohortToUserIdList().entrySet()) {
+        String cohortName = ele.getKey();
+        List<String> users = ele.getValue();
+        if (!this.cohortUserMapper.containsKey(cohortName)) {
+          this.cohortUserMapper.put(cohortName, new CohortWSStr());
+        }
+        this.cohortUserMapper.get(cohortName).addCubletResults(users);
+      }
+      this.result.clearUserIds();
     }
     return this.result;
   }
 
+  /**
+   * Persist cohort to output disk.
+   *
+   * @param outputDir the output file path
+   */
+  public void persistCohort(String outputDir) throws IOException {
+    for (Map.Entry<String, CohortWSStr> ele : this.cohortUserMapper.entrySet()) {
+      String fileName = ele.getKey() + ".cohort";
+      File cubemeta = new File(outputDir, fileName);
+      DataOutputStream out = new DataOutputStream(new FileOutputStream(cubemeta));
+      ele.getValue().writeTo(out);
+    }
+  }
+
+  /**
+   * Persist cohort to output disk.
+   *
+   * @param cohortPath the path to store the previous stored cohort.
```

Comment:
Please add the structure of the cohort storage (i.e., fold level) in this annotation. Thanks

Reply:
added

Reply:
please add an example cohort file name into the annotation.

Reply:
added